gnome-initial-setup/pages/branding-welcome/gis-branding-welcome-page.c

```diff
@@ -0,0 +1,289 @@
+   * Pango failed so far for some reason, so I'm hardcoding this for now. */
+  attr = pango_attr_rise_new (6 * PANGO_SCALE);
+
+  pango_attr_list_insert (attr_list,attr);
+  gtk_label_set_attributes (GTK_LABEL (WID ("branding-text")), attr_list);
+  pango_attr_list_unref (attr_list);
+}
+
+static void
+gis_branding_welcome_page_constructed (GObject *object)
+{
+  GisBrandingWelcomePage *page = GIS_BRANDING_WELCOME_PAGE (object);
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+
+  G_OBJECT_CLASS (gis_branding_welcome_page_parent_class)->constructed (object);
+
+  gtk_container_add (GTK_CONTAINER (page), WID ("branding-welcome-page"));
+
+  update_branding_specific_info (page);
+  update_line_height_for_description (page);
+  load_css_overrides (page);
+
+  gis_page_set_complete (GIS_PAGE (page), TRUE);
+  gtk_widget_show (GTK_WIDGET (page));
+}
+
+static void
+gis_branding_welcome_page_finalize (GObject *object)
+{
+  GisBrandingWelcomePage *page = GIS_BRANDING_WELCOME_PAGE (object);
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+
+  g_free (priv->title);
+  g_free (priv->description);
+  g_free (priv->logo_path);
+  g_free (priv->skip_pages);
+
+  G_OBJECT_CLASS (gis_branding_welcome_page_parent_class)->finalize (object);
+}
+
+static void
+gis_branding_welcome_page_class_init (GisBrandingWelcomePageClass *klass)
+{
+  GisPageClass *page_class = GIS_PAGE_CLASS (klass);
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  page_class->page_id = PAGE_ID;
+  object_class->constructed = gis_branding_welcome_page_constructed;
+  object_class->finalize = gis_branding_welcome_page_finalize;
+}
+
+static void
+gis_branding_welcome_page_init (GisBrandingWelcomePage *page)
+{
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+
+  priv->title = NULL;
+  priv->description = NULL;
+  priv->logo_path = NULL;
+  priv->skip_pages = NULL;
+
+  g_resources_register (branding_welcome_get_resource ());
+}
+
+void
+gis_prepare_branding_welcome_page (GisDriver *driver)
+{
+  GisPage *page = NULL;
+  GisBrandingWelcomePagePrivate *priv = NULL;
+
+  if (gis_driver_is_skipped_page (driver, PAGE_ID))
+    return;
+
+  page = GIS_PAGE (g_object_new (GIS_TYPE_BRANDING_WELCOME_PAGE, "driver", driver, NULL));
+  priv = gis_branding_welcome_page_get_instance_private (GIS_BRANDING_WELCOME_PAGE (page));
+
+  /* We don't actually add the page unless we have the only required field */
+  if (priv->title == NULL) {
+    g_object_ref_sink (page);
+    g_object_unref (page);
+    return;
+  }
+
+  if (priv->skip_pages != NULL) {
+    g_auto(GStrv) skip_pages = g_strsplit (priv->skip_pages, ",", -1);
+    gint i;
+
+    for (i = 0; skip_pages[i]; i++) {
+      g_debug ("Skipping page %s...", skip_pages[i]);
+      gis_driver_add_skipped_page (driver, skip_pages[i]);
+    }
+  }
```

Comment:
Could this not be done from the constructed implementation, after the configuration has been loaded?